### PR TITLE
Safe mint update

### PR DESCRIPTION
**Pull request by reza2732**
Updated the ordering of safeMint to mint before making a ERC721Receiver external call.

### CODE REVIEW

**Comment by jacobd** [comment]
I think you're right but line 367 should be
`_checkOnERC721Received(address(0), to, _tokens.length - 1, _data),`
because _mint() pushes on to the _tokens array.

File: contracts/ERC721S.sol [hunk 1]
```diff
@@ -362,11 +362,11 @@ contract ERC721Sequential is Context, ERC165, IERC721, IERC721Metadata {
      * forwarded in {IERC721Receiver-onERC721Received} to contract recipients.
      */
     function _safeMint(address to, bytes memory _data) internal virtual {
+        _mint(to);
         require(
             _checkOnERC721Received(address(0), to, _tokens.length, _data),
```
Comment (jacobd) [inline contracts/ERC721S.sol:367]:
```suggestion
            _checkOnERC721Received(address(0), to, _tokens.length - 1, _data),
```

**Review by jacobd — COMMENTED** [review]
since _mint() pushes on to the array the tokenId is the length - 1

**Comment by reza2732** [comment]
@jacobd Nice catch!

**Review by detour1999 — APPROVED** [review]
LGTM!

**Pull request merged**